core/lib/spree/preferences/static_model_preferences.rb

```diff
@@ -36,9 +36,14 @@ def initialize
       end
 
       def add(klass, name, preferences)
+        constantized_klass = klass.try(:constantize) || klass
```

Comment:
This will trigger an autoload during initialization, which is one of the things that this change should avoid. Thoughts on just converting to a string if it's not already and allowing `Definition` to accept it as well?

Reply:
Hey @elia, I'm wondering how to fix it. In the end, `add` creates another `Definition` instance, and [it calls `.defined_preferences` on the class](https://github.com/solidusio/solidus/blob/c914bd1abf47029d9295e656c9d4017c34c2d12b/core/lib/spree/preferences/static_model_preferences.rb#L12) :shrug: